### ISSUE TITLE
Negative enum variants parse to heap pointers?

### ISSUE THREAD

**Issue by nshp** [Vector35/binaryninja-api#1810]
When I hit `i` in the types view and define an enum like:
```c
enum foo {
    FOO_A = -2,
    FOO_B = -1,
    FOO_C = 0,
    FOO_D = 1
};
```

The resulting enum is:
```c
enum foo
{
    FOO_A = 0xffffaa2453b0faa0,
    FOO_B = 0xffffaa2453b0faa0,
    FOO_C = 0x0,
    FOO_D = 0x1,
};
```

The bitwise not of `0xffffaa2453b0faa0` is `0x55dbac4f055f`, which suspiciously lines up with:
```
/proc/$(pidof binaryninja)/maps:
55db95c8d000-55dc30f2d000 rw-p 00000000 00:00 0        [heap]
```

**Comment by plafosse** [comment]
Ha, yeah so it does, bad parser rule. I have a fix for it already.

**Comment by plafosse** [comment]
Fixed in 2.1.2268
